R/audits.R

```diff
@@ -1,110 +1,49 @@
-get_field <- function(apps, field, include_null = FALSE) {
-  all <- lapply(apps, function(x) {
-    x[[field]]
-  })
-  empty <- sapply(all, is.null)
-
-  if (!include_null) {
-    return(all[!empty])
-  }
-  all
-}
-
-#' Get information on all apps for a server
-#'
-#' \lifecycle{experimental}
-#'
-#' @param connect A Connect object
-#'
-#' @return List with application data, to be used by audit functions
-#' @family audit functions
-#' @export
-cache_apps <- function(connect) {
-  apps <- connect$get_apps()
-  apps
-}
-
-#' Audit Vanity URLs
+#' Check to see if a vanity URL is currently in use
 #'
 #' \lifecycle{experimental}
 #'
-#' @param apps App list, see [cache_apps()]
-#' @param server_url Base url for the Connect server
-#' @param vanity Optional, see details
+#' @param connect A Connect R6 object
+#' @param vanity string of the vanity URL to check
 #'
-#' @details If `vanity` is not provided, returns a list of all the vanity
-#'   urls in use. If `vanity` is provided, returns whether or not
-#'   `vanity` is eligible as a vanity url.
+#' @return logical indicating if the vanity URL is available.
 #'
 #' @family audit functions
 #' @export
-audit_vanity_urls <- function(apps, server_url, vanity = NULL) {
-  # TODO: why does vanities not work?
-  urls <- get_field(apps, "url")
-  parse_server <- httr::parse_url(server_url)
-  if (is.null(parse_server$scheme)) {
-    stop(glue::glue("ERROR: protocol (i.e. http:// or https://) not defined on server_url={server_url}"))
-  }
-  content <- sapply(urls, function(u) {
-    trim_vanity(u, parse_server$path)
-  })
+vanity_is_available <- function(connect, vanity) {
```

Comment:
This might be a server question more than anything, but what happens to characters that need url encoding? Do they come back form the server encoded (such that we should URL encode here before checking availability?)

Reply:
Good question, I'm not sure.

Reply:
I tried to add a vanity url with spaces in the web app and it errored, so I think this is fine.

![image](https://github.com/rstudio/connectapi/assets/2975928/17bfef58-a02d-4bfc-928f-413212cfe6bd)